### PR TITLE
[now-next] Add support for experimental basePath

### PR DESCRIPTION
**Pull request by ijjk**
This adds support for https://github.com/zeit/next.js/pull/9872. ~**Note**: the added tests won't pass until the below PR is shipped on canary~ 

relies on: https://github.com/zeit/next.js/pull/9875

### CODE REVIEW

**Review by timneutkens — APPROVED** [review]

File: packages/now-next/src/utils.ts [hunk 1]
```diff
@@ -308,6 +308,7 @@ type RoutesManifestRegex = {
 };
 
 export type RoutesManifest = {
+  basePath: string;
```
Comment (timneutkens) [inline packages/now-next/src/utils.ts]: This can be undefined on older versions of next.

```suggestion
  basePath: string | undefined;
```

Comment (ijjk) [reply packages/now-next/src/utils.ts]: Updated

File: packages/now-next/src/index.ts [hunk 2]
```diff
@@ -336,13 +336,17 @@ export const build = async ({
...
+  let nextBasePath: string | undefined;
 
   if (routesManifest) {
     switch (routesManifest.version) {
       case 1:
       case 2: {
         redirects.push(...convertRedirects(routesManifest.redirects));
         rewrites.push(...convertRewrites(routesManifest.rewrites));
+        if (routesManifest.basePath) {
+          nextBasePath = routesManifest.basePath;
```
Comment (Timer) [inline packages/now-next/src/index.ts:349]: We should add an `invariant` here that the `basePath` starts with a `/` and does not end with a trailing `/`. Also, the value `/` should be special cased and ignored as that's the default.

File: packages/now-next/src/index.ts [hunk 3]
```diff
@@ -400,6 +404,17 @@ export const build = async ({
       routes: [
         // TODO: low priority: handle trailingSlash
 
+        // Add top level rewrite for basePath if provided
+        ...(nextBasePath
+          ? [
+              {
+                src: `${nextBasePath}/(.*)`,
```
Comment (Timer) [inline packages/now-next/src/index.ts]: Does this work for someone visiting just `<basePath>` (think about no trailing slash)? Should this be the following?
```suggestion
                src: `^${nextBasePath}(?:$|/(.*))$`,
```

Comment (Timer) [reply packages/now-next/src/index.ts]: ☝️ 

image

File: packages/now-next/src/index.ts [hunk 4]
```diff
@@ -941,6 +956,17 @@ export const build = async ({
       ...staticDirectoryFiles,
     },
     routes: [
+      // Add top level rewrite for basePath if provided
+      ...(nextBasePath
+        ? [
+            {
+              src: `${nextBasePath}/(.*)`,
```
Comment (Timer) [inline packages/now-next/src/index.ts]: Ditto from above.

File: packages/now-next/test/fixtures/16-base-path/now.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  "builds": [{ "src": "package.json", "use": "@now/next" }],
+  "probes": [
+    {
+      "path": "/docs/_next/static/testing-build-id/pages/index.js",
+      "responseHeaders": {
+        "cache-control": "public,max-age=31536000,immutable"
+      }
+    },
+    {
+      "path": "/docs/",
```
Comment (Timer) [inline packages/now-next/test/fixtures/16-base-path/now.json:12]: Test the non-slash version of this per my above suggestion.

**Review by Timer — DISMISSED** [review]

File: packages/now-next/src/index.ts (hunk 2 above)
Comment (ijjk) [reply packages/now-next/src/index.ts:349]: Added some invariants and special cased the default `/`

File: packages/now-next/src/index.ts (hunk 3 above)
Comment (ijjk) [reply packages/now-next/src/index.ts]: Updated

File: packages/now-next/test/fixtures/16-base-path/now.json (hunk 5 above)
Comment (ijjk) [reply packages/now-next/test/fixtures/16-base-path/now.json:12]: Added additional check

**Review by timneutkens — APPROVED** [review]

**Pull request merged**